### PR TITLE
adaptive black frame search range

### PR DESCRIPTION
**Pull request by rlauuzo**

### CODE REVIEW

**Comment by rlauuzo** [comment]
This now leverages the credit lengths from previous episodes to optimize subsequent scans within the same season. And provides a decent improvement in speed.

**Comment by AbandonedCart** [comment]
What's the basis for 0.7 and 1.5 (initial start and intro is null)? Just curious

File: ConfusedPolarBear.Plugin.IntroSkipper/Analyzers/BlackFrameAnalyzer.cs [hunk 1]
```diff
@@ -49,14 +63,20 @@ public BlackFrameAnalyzer(ILogger<BlackFrameAnalyzer> logger)
 
             var intro = AnalyzeMediaFile(
```
Comment (AbandonedCart) [inline ConfusedPolarBear.Plugin.IntroSkipper/Analyzers/BlackFrameAnalyzer.cs:64]: We really need to change "intro" to something else. I always think it's the actual intro at first glance.

**Review by AbandonedCart — APPROVED** [review]
I had to double check that black frame instances are per season and it is, so that works.

**Comment by rlauuzo** [comment]
> What's the basis for 0.7 and 1.5 (initial start and intro is null)? Just curious

They were just placeholders. I've improved it by replacing them with a method that calculates the starting point for the first episode.

**Pull request merged**